tools/run_tests/xds_k8s_test_driver/bin/cleanup/cleanup.py

```diff
@@ -0,0 +1,374 @@
+                logging.info(
+                    'Skipped [keep]: GCE resource suffix [%s] is marked as keep',
+                    result.group(1))
+                continue
+            remove_relative_resources_run_xds_tests(project, result.group(1))
+            continue
+
+        # Cleanup resources from the gke framework.
+        result = re.search(f'{PSM_SECURITY_PREFIX}-health-check-(.*)',
+                           resource['name'])
+        if result is not None:
+            if is_marked_as_keep_gke(result.group(1)):
+                logging.info(
+                    'Skipped [keep]: GKE resource suffix [%s] is marked as keep',
+                    result.group(1))
+                continue
+            cleanup_td_for_gke(project, network, PSM_SECURITY_PREFIX,
+                               result.group(1))
+            continue
+
+        logging.info(
+            '----- Skipped [does not matching resource name templates]')
+
+    # Delete leaked instance templates, those usually mean there are leaked VMs
+    # from the gce framework. Also note that this is only needed for the gce
+    # resources.
+    leakedInstanceTemplates = exec_gcloud(project, 'compute',
+                                          'instance-templates', 'list')
+    for resource in leakedInstanceTemplates:
+        logger.info('-----')
+        logger.info('----- Cleaning up instance template %s', resource['name'])
+        if dry_run:
+            # Skip deletion for dry-runs
+            logging.info('----- Skipped [Dry Run]: %s', resource['name'])
+            continue
+
+        # Cleanup resources from the gce framewok.
+        result = re.search(r'test-template(.*)', resource['name'])
+        if result is not None:
+            if is_marked_as_keep_gce(result.group(1)):
+                logging.info(
+                    'Skipped [keep]: GCE resource suffix [%s] is marked as keep',
+                    result.group(1))
+                continue
+            remove_relative_resources_run_xds_tests(project, result.group(1))
+            continue
+
+        logging.info(
+            '----- Skipped [does not matching resource name templates]')
+
+    # Delete leaked k8s namespaces, those usually mean there are leaked testing
+    # client/servers from the gke framework.
+    k8s_api_manager = k8s.KubernetesApiManager(xds_k8s_flags.KUBE_CONTEXT.value)
+    nss = k8s_api_manager.core.list_namespace()
+    for ns in nss.items:
+        logger.info('-----')
+        logger.info('----- Cleaning up k8s namespaces %s', ns.metadata.name)
+        if ns.metadata.creation_timestamp <= utc.localize(
+                get_expire_timestamp()):
+            if dry_run:
+                # Skip deletion for dry-runs
+                logging.info('----- Skipped [Dry Run]: %s', ns.metadata.name)
+                continue
+
+            result = re.search(f'{PSM_SECURITY_PREFIX}-server-(.*)',
+                               ns.metadata.name)
```

Comment:
We could use 2 for-loop here, one looping "server"/"client", another looking through PSM_SECURITY_PREFIX/URL_MAP_TEST_PREFIX. Saves a couple LOC.